shardingsphere-sql-parser/shardingsphere-sql-parser-dialect/shardingsphere-sql-parser-postgresql/src/main/antlr4/imports/postgresql/DDLStatement.g4

```diff
@@ -17,34 +17,187 @@
 
 grammar DDLStatement;
 
-import Symbol, Keyword, PostgreSQLKeyword, Literals, BaseRule;
+import Symbol, Keyword, PostgreSQLKeyword, Literals, BaseRule,DMLStatement;
 
 createTable
-    : CREATE createTableSpecification_ TABLE tableNotExistClause_ tableName createDefinitionClause inheritClause_
+    : CREATE createTableSpecification_ TABLE tableNotExistClause_ tableName
+      createDefinitionClause
+      (OF anyName optTypedTableElementList)?
+      (PARTITION OF qualifiedName optTypedTableElementList partitionBoundSpec)?
+      inheritClause_ optPartitionSpec_ tableAccessMethodClause optWith onCommitOption optTableSpace
+      (AS select optWithData)?
+      (EXECUTE name executeParamClause optWithData)?
+    ;
+
+executeParamClause
+    : LP_ exprList RP_
+    ;
+
+partitionBoundSpec
+    : FOR VALUES WITH LP_ hashPartbound RP_
+    | FOR VALUES IN LP_ exprList RP_
+    | FOR VALUES FROM LP_ exprList RP_ TO LP_ exprList RP_
+    | DEFAULT
+    ;
+
+hashPartbound
+    : hashPartboundElem
+    | hashPartbound COMMA_ hashPartboundElem
+    ;
+
+hashPartboundElem
+    : nonReservedWord NUMBER_
+    ;
+
+optTypedTableElementList
+     : LP_ typedTableElementList RP_
+     ;
+
+typedTableElementList
+    : typedTableElement
+    | typedTableElementList COMMA_ typedTableElement
+    ;
```

Comment:
Maybe, it is good like that
```
typedTableElementList
    : typedTableElement*
    ;
```
it can reduce recursion and looks simple. it is same as others.